File: 프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.java
import java.util.*;
class Solution {
    public int[] solution(int n) {
        int[] answer = new int[n * (n+1) / 2];
        int num = 1;
        ArrayList<int[]> triangle = new ArrayList<>();
        
        for(int i = 0 ; i < n ; i++){
            triangle.add(new int[i+1]);
        }
        int x = -1 , y = 0;
        for(int i = 0 ; i<n ; i++){
            for(int j = i ; j<n ;j++){
                
                if(i % 3 == 0) x++;
                else if(i % 3 == 1) y++;
                else{
                    x--;
                    y--;
                }
                triangle.get(x)[y] = num++;
            }
        }
        int k = 0;
        for(int[] row : triangle){
            for(int val : row){
                answer[k++] = val;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/68645. 삼각 달팽이/삼각 달팽이.js
function solution(n) {
    // 삼각형을 저장할 2차원 배열을 초기화합니다.
    const triangle = Array.from({ length: n }, (_, i) => Array(i + 1).fill(0));
    console.log(triangle)
    let num = 1; // 삼각형에 채울 숫자
    let x = -1, y = 0; // 시작 위치

    for (let i = 0; i < n; i++) {
        for (let j = i; j < n; j++) {
            if (i % 3 === 0) {
                // 아래로 이동
                x++;
            } else if (i % 3 === 1) {
                // 오른쪽으로 이동
                y++;
            } else if (i % 3 === 2) {
                // 위로 이동
                x--;
                y--;
            }
            triangle[x][y] = num++;
        }
    }

    // 2차원 배열을 1차원 배열로 변환하여 반환합니다.
    return triangle.flat();
}


